packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+
+const double _kMinFlingVelocity = 1.0;  // screen width per second
+
+// Used for iOS.
+class CupertinoPageTransition extends AnimatedWidget {
+  static final FractionalOffsetTween _kTween = new FractionalOffsetTween(
+    begin: FractionalOffset.topRight,
+    end: -FractionalOffset.topRight
+  );
+
+  CupertinoPageTransition({
+    Key key,
+    Animation<double> animation,
+    this.child
+  }) : super(
+    key: key,
+    listenable: _kTween.animate(new CurvedAnimation(
+      parent: animation,
+      curve: new _CupertinoTransitionCurve(null)
+    )
+  ));
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO(ianh): tell the transform to be un-transformed for hit testing
+    // but not while being controlled by a gesture.
+    return new SlideTransition(
+      position: listenable,
+      child: new PhysicalModel(
+        shape: BoxShape.rectangle,
+        color: const Color(0x00000000), // we don't need the canvas background, just the elevation
```

Comment:
The shadow is cast by the PhysicalModel rectangle.

This code seems wrong to me. I don't understand why the background would be explicitly given but be transparent. That's almost always a sign that something is wrong.